Mesh_3/doc/Mesh_3/Concepts/InitialPointsGenerator.h

```diff
@@ -0,0 +1,64 @@
+/*!
+\ingroup PkgMesh3SecondaryConcepts
+\cgalConcept
+
+The function object concept `InitialPointsGenerator` is designed to construct
+a set of initial points on the surface of the domain.
+
+\cgalHasModelsBegin
+\cgalHasModels{CGAL::Construct_initial_points_labeled_image}
+\cgalHasModels{CGAL::Construct_initial_points_gray_image}
+\cgalHasModelsEnd
+
+*/
+
+class InitialPointsGenerator {
+public:
+
+/// \name Operations
+/// @{
+
+/*!
+Output a set of (`n`) surface points to the
+output iterator `pts`, as objects of type
+`std::tuple<Weighted_point_3, int, Index>`.
+`Weighted_point_3` is the point's position and weight,
+`int` is the dimension of the minimal dimension subcomplex on which the point lies, and
+`Index` is the underlying subcomplex index.
+
+@tparam OutputIterator model of `OutputIterator`, containing points of type
+`std::tuple<C3t3::Triangulation::Geom_traits::Weighted_point_3, int, MeshDomain::Index>`
+@tparam MeshDomain model of `MeshDomain_3`
+@tparam C3t3 model of `MeshComplex_3InTriangulation_3`
+@param pts the output points
+@param domain the input domain
+@param c3t3 the input complex
+@param n an estimation of the number of points to output
```

Comment:
In the original code that add points in labeled images by looking at connected components, `n` is used as the number of rays launched per seed. If a ray does not intersect a surface, then no point is added.
So it is more like a quality parameter, and the number of points outputted should be roughly `n`, it can be more, or it can be less.

Reply:
I know :-)
The doc should give a little more info though. An external user would wonder what "estimation" means, for a parameter